R/bainCommon.R

```diff
@@ -0,0 +1,1288 @@
+        }
+      } else if (options$bayesFactorType == "BF10") {
+        if (options$hypothesis == "equalNotEqual") {
+          row <- list(Variable=variable, "hypothesis[type1]" = gettext("H0: Equal"),"BF[type1]"="", "pmp[type1]" = tableResults[["PMP_0"]],
+                      "hypothesis[type2]" = gettext("H1: Not equal"), "BF[type2]" = tableResults[["BF_0u"]], "pmp[type2]" = tableResults[["PMP_u"]])
+        } else if (options$hypothesis == "equalBigger") {
+          row <-list(Variable=variable, "hypothesis[type1]" = gettext("H0: Equal"),"BF[type1]"="", "pmp[type1]" = tableResults[["PMP_0"]],
+                     "hypothesis[type2]" = gettext("H1: Bigger"), "BF[type2]" = tableResults[["BF_01"]], "pmp[type2]" = tableResults[["PMP_1"]])
+        } else if (options$hypothesis == "equalSmaller") {
+          row <-list(Variable=variable, "hypothesis[type1]" = gettext("H0: Equal"), "BF[type1]"="", "pmp[type1]" = tableResults[["PMP_0"]],
+                     "hypothesis[type2]" = gettext("H1: Smaller"), "BF[type2]" = tableResults[["BF_01"]], "pmp[type2]" = tableResults[["PMP_1"]])
+        } else if (options$hypothesis == "biggerSmaller") {
+          row <-list(Variable=variable, "hypothesis[type1]" = gettext("H1: Bigger"), "BF[type1]"= "", "pmp[type1]" = tableResults[["PMP_0"]],
+                     "hypothesis[type2]" = gettext("H2: Smaller"), "BF[type2]" = tableResults[["BF_01"]], "pmp[type2]" = tableResults[["PMP_1"]])
+        } else if (options$hypothesis == "equalBiggerSmaller") {
+          row <-list(Variable=variable, "type[equal]" = gettext("H0: Equal"), "BF[equal]"= "", "pmp[equal]" = tableResults[["PMP_0"]],
+                     "type[greater]"= gettext("H1: Bigger"), "BF[greater]" = tableResults[["BF_01"]], "pmp[greater]" = tableResults[["PMP_1"]],
+                     "type[less]" = gettext("H2: Smaller"), "BF[less]" = tableResults[["BF_02"]], "pmp[less]" = tableResults[["PMP_2"]])
+        }
+      }
+      table$addRows(row, rowNames = variable)
+      progressbarTick()
+    }
+  } else if(type == "pairedTTest"){
+    testType <- base::switch(options[["hypothesis"]],
+                             "equalNotEqual"       = 1,
+                             "equalBigger"         = 2,
+                             "equalSmaller"        = 3,
+                             "biggerSmaller"       = 4,
+                             "equalBiggerSmaller"  = 5)
+    
+    table$setExpectedSize(length(options[["pairs"]]))
+    
+    startProgressbar(length(options[["pairs"]]))
+    
+    for (pair in options[["pairs"]]){
+      
+      currentPair <- paste(pair, collapse=" - ")
+      
+      if(pair[[1]] != "" || pair[[2]] != ""){
+        
+        bainResult <- .bainGetGeneralTestResults(dataset, options, bainContainer, ready, type, pair = pair, testType = testType)
+        
+        if (isTryError(bainResult)) {
```

Comment:
`bainGetGeneralTestResults()` returns the results from `bainGetPairedTTestResults()` which has the following error handling:
```
    p <- try({
      bainResult <- .bainTTestRaw(options, x = c1, y = c2, type = testType, paired = TRUE)
    })

    if(isTryError(p)){
      bainContainer$setError(gettextf("An error occurred in the analysis:<br>%1$s.", jaspBase::.extractErrorMessage(p)))
      return()
    }
```
which means that `bainResult` is `NULL` and not a `try-error`.
I think you want to check the container here (assuming you have a container per pair, otherwise one pair would wreck all analyses)